lib/cocina/generator/schema_base.rb

```diff
@@ -63,8 +63,8 @@ def relaxed_comment
       def dry_datatype(doc)
         return doc.name if doc.name.present? && schemas.include?(doc.name)
 
-        datatype_from_doc_type(doc) ||
-          datatype_from_doc_names(doc) ||
+        datatype_from_doc_names(doc) ||
```

Comment:
This re-ordering lets us simplify the code in this class (see `#datatype_from_doc_names` below) and does not break or otherwise change models in unexpected ways.

Reply:
Generation is suuuuuper fast now.

Reply:
execution time, not generation time.  Like, when instantiating a cocina object from JSON or creating one from code or whatever.  Again ... prolly doesn't matter

Reply:
AFAIK, this code is only called when generating.